Userland/Libraries/LibWeb/Layout/FlexFormattingContext.cpp

```diff
@@ -1179,28 +1179,37 @@ void FlexFormattingContext::distribute_any_remaining_free_space()
 
         // 12.2.
         float space_between_items = 0;
-        float space_before_first_item = 0;
+        float initial_offset = 0;
         auto number_of_items = flex_line.items.size();
 
+        enum Direction {
+            LeftToRight,
+            RightToLeft
+        };
+        int direction = Direction::LeftToRight;
+
         switch (flex_container().computed_values().justify_content()) {
         case CSS::JustifyContent::FlexStart:
+            initial_offset = 0;
+            break;
         case CSS::JustifyContent::FlexEnd:
-            space_before_first_item = 0;
+            direction = Direction::RightToLeft;
+            initial_offset = m_available_space->main.value_or(NumericLimits<float>::max());
             break;
         case CSS::JustifyContent::Center:
-            space_before_first_item = (m_available_space->main.value_or(NumericLimits<float>::max()) - used_main_space) / 2.0f;
+            initial_offset = (m_available_space->main.value_or(NumericLimits<float>::max()) - used_main_space) / 2.0f;
             break;
         case CSS::JustifyContent::SpaceBetween:
             space_between_items = flex_line.remaining_free_space / (number_of_items - 1);
             break;
         case CSS::JustifyContent::SpaceAround:
             space_between_items = flex_line.remaining_free_space / number_of_items;
-            space_before_first_item = space_between_items / 2.0f;
+            initial_offset = space_between_items / 2.0f;
             break;
         }
 
         // FIXME: Support reverse
```

Comment:
Isn't this what these changes fix? If so, remove this `FIXME`.

Reply:
I believe so therefore I'll drop the FIXME and add a comment explaining how it works